enhancements/installer/aws-internal-clusters.md

```diff
@@ -0,0 +1,192 @@
+
+No public subnets are required, since no public Load Balancer will be created. And, since no public records will be required, the current requirement for the existence of a public R53 Zone matching the `BaseDomain` will no longer be required. The installer still creates the private R53 Zone for the cluster.
+
+This affects the default ingress setup for the cluster to be internal too.
+
+### User Stories
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints
+
+#### API
+
+A new field `publish` is added to the InstallConfig. Publish controls how the user facing endpoints of the cluster like the Kuberenets API, OpenShift Ingress etc. are exposed. Valid values are `External` (the default) and `Internal`.
+
+#### Resource provided to the installer
+
+The users provide a list of private subnets to the installer and set the `publish` to `Internal`.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+publish: Internal
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+#### Basedomain
+
+The basedomain will continue to be used for creating the private Route53 zone for the cluster and all the necessary records but, the requirement that a public Route53 Zone exist corresponding to the basedomain will **NOT** be needed anymore.
+
+The CustomResource [`DNSes.config.openshift.io`][openshift-api-config-dns] `cluster` object's `.spec.publicZone` will be set to empty. This will make sure that the operators do not create any public records for the cluster as per the API [reference][openshift-api-config-dns-empty-public-zone]
+
+#### Public Subnets
+
+No public subnets need to be provided to the installer.
+
+#### Access to Internet
+
+The cluster still continues to require access to Internet.
+
+#### Resources created by the installer
+
+The installer will no longer create (vs. the fully-IPI flow with pre-existing VPC):
+
+- Public Load Balancers for API (6443) (aws_lb.api_external, aws_lb_target_group.api_external, aws_lb_listener.api_external_api)
+- No security group that allows 6443 from Internet.
+- Public DNS record (aws_route53_record.api_external)
+- No public IP address associated to bootstrap-host (associate_public_ip_address: false)
+- No security group that allows SSH to bootstrap-host from Internet (aws_security_group_rule.ssh)
+
+The installer will continue to create (vs. the fully-IPI flow with pre-existing VPC):
+
+- Private Load Balancers for API (6443) and MCS (22623)
```

Comment:
Nit: "API" -> "Kubernetes API" and "MCS" -> "machine-config server"

Reply:
done https://github.com/openshift/enhancements/compare/d1df0a7526e3ead64c786034f1f50e29d556a5aa..d5621ab1053e0b04209359ebcd82d867ab38ac2c